kq_epic_post.md

```diff
@@ -0,0 +1,160 @@
+
+And yet, the wraiths were still at our heels. Due to the multiple iterations of bug fixing, we had to run more transitions to update pull requests records. In one, we discovered that our use of Active Record lookups allowed for a gap between reading and writing and that direct queries were needed. In another, we learned how to be more targeted by creating a CSV of mismatched pull requests and iterating over them directly.
+
+### Revel in your triumph
+
+Though the quest was arduous and many shadows befell us, at last we reaped the harvest of victory. (_i hope :lolsob:_)
+
+In this epic as we ran 12 [Scientist](https://github.com/github/scientist) experiments to validate our changes, some with multiple iterations.
+
+_Results with ✨graphs✨_
+
+### Reflect on your journey
+
+This project spanned many months and thus there was a lot to learn.
+1. Even if there are only a few people directly working on it at first, start a new public Slack channel dedicated solely to the project. This centralizes conversations and makes it easier to find history and share with others.
+2. Communicate with other teams early if the work could span beyond your team. As these queries occur on a shared `issues-pull-requests` cluster, eventually the Issues team began work that overlapped with ours. If we'd started that conversation earlier we could have more quickly identified opportunities to collaborate.
+3. Be careful with your feature flags. When we were troubleshooting the dual writing bugs, we opened a pull request to implement a new pathway while also removing the old way. On the surface that seemed like an efficient way to do it, but what we didn't consider was that the two ways were controlled by different feature flags. So there was a gap between when the old one was turned off and the new one was fully ramped up. This was a contributing factor to our record mismatches.
+   
+Additionally, this project generated some open questions for future development.
+1. Currently, the queries section of a service's scorecard only passes if there are no killed/slow queries. Even one causes it to fail. Is there a way that we can make this more granular to better estimate impact and priority?
+2. Finding and verifying database records was difficult and left us with low confidence. Is there a better way to identify which source to use, how, and when?
+3. Transitions can be tricky and we encountered several setbacks. How can we improve this process to allow for better troubleshooting?
```

Comment:
omg yes it's been years since my last transition but I feel this a lot. Transitions are so demanding to perform; they are full of manual steps which means they're prone to human mistakes, and they require so much sustained executive function to see through. literally months! that's risky.